--- a/pokemonsay.sh
+++ b/pokemonsay.sh
@@ -36,7 +36,8 @@
 
 INSTALL_PATH=${HOME}/.bin/pokemonsay
 # Where the pokemon are.
-POKEMON_PATH=${INSTALL_PATH}/pokemons
+POKEMON_PATH=${INSTALL_PATH}/pokemonsy
+COWPATH=${INSTALL_PATH}/pokemonsy
 
 list_pokemon() {
 	echo "Pokemon available in '$POKEMON_PATH/':"
@@ -55,7 +56,7 @@
 list_pokemon_forms() {
 	echo "Pokemon available in '$POKEMON_PATH/':"
 	echo
-	ALL_POKEMON="$(find $POKEMON_PATH -regextype sed -regex ".*[\-]\+.*\.cow" | sort)"
+	ALL_POKEMON="$(find $POKEMON_PATH -printf "%f" -regextype sed -regex ".*[\-]\+.*\.cow" | sort)"
 	echo "$ALL_POKEMON" | while read -r POKEMON; do
 		POKEMON=${POKEMON##*/}
 		POKEMON=${POKEMON%.cow}
@@ -67,69 +68,73 @@
 }
 
 # While there are arguments, keep reading them.
-while [ $# -gt 0 ]
-do
-key="$1"
-case $key in
-	-p|--pokemon)
+while [ $# -gt 0 ]; do
+	key="$1"
+	case $key in
+	-p | --pokemon)
 		POKEMON_NAME="$2"
-		shift; shift
+		shift
+		shift
 		;;
-	-p=*|--pokemon=*)
+	-p=* | --pokemon=*)
 		POKEMON_NAME="${1#*=}"
 		shift
 		;;
-	-d|--nationalDex)
+	-d | --nationalDex)
 		NATIONAL_DEX="$2"
-		shift; shift
+		shift
+		shift
 		;;
-	-d=*|--nationalDex=*)
+	-d=* | --nationalDex=*)
 		NATIONAL_DEX="${1#*=}"
 		shift
 		;;
-	-D|--nationalDexForms)
+	-D | --nationalDexForms)
 		NATIONAL_DEX_FORMS="$2"
-		shift; shift
+		shift
+		shift
 		;;
-	-D=*|--nationalDexForms=*)
+	-D=* | --nationalDexForms=*)
 		NATIONAL_DEX_FORMS="${1#*=}"
 		shift
 		;;
-	-f|--file)
+	-f | --file)
 		COW_FILE="$2"
-		shift; shift
+		shift
+		shift
 		;;
-	-f=*|--file=*)
+	-f=* | --file=*)
 		COW_FILE="${1#*=}"
 		shift
 		;;
-	-W|--word-wrap)
+	-W | --word-wrap)
 		WORD_WRAP="$2"
-		shift; shift
+		shift
+		shift
 		;;
-	-W=*|--word-wrap=*)
+	-W=* | --word-wrap=*)
 		WORD_WRAP="${1#*=}"
 		shift
 		;;
-	-n|--no-wrap)
+	-n | --no-wrap)
 		DISABLE_WRAP="YES"
 		shift
 		;;
-	-l|--list)
+	-l | --list)
 		list_pokemon
 		;;
-	-L|--list-forms|--listForms)
+	-L | --list-forms | --listForms)
 		list_pokemon_forms
 		;;
-	-N|--no-name)
+	-N | --no-name)
 		DISPLAY_NAME="NO"
 		shift
 		;;
-	-t|--think)
+	-t | --think)
 		THINK="YES"
 		shift
 		;;
-	-h|--help)
+	-h | --help)
 		usage
 		;;
 	-*)
@@ -146,7 +151,7 @@
 		fi
 		shift
 		;;
-esac
+	esac
 done
 
 # Disable wrapping if the option is set, otherwise
@@ -159,16 +164,16 @@
 fi
 
 # Define which pokemon should be displayed.
-if   [ -n "$POKEMON_NAME" ]; then
-	POKEMON_COW=$(find $POKEMON_PATH -iname "*$POKEMON_NAME.cow" | shuf -n 1)
+if [ -n "$POKEMON_NAME" ]; then
+	POKEMON_COW=$(find $POKEMON_PATH -printf "%f" -iname "*$POKEMON_NAME.cow" | shuf -n 1)
 elif [ -n "$NATIONAL_DEX" ]; then
-	POKEMON_COW=$(find $POKEMON_PATH -regextype sed -regex ".*${NATIONAL_DEX}\_\w*[^\-]\.cow" | shuf -n 1)
+	POKEMON_COW=$(find $POKEMON_PATH -printf "%f" -regextype sed -regex ".*${NATIONAL_DEX}\_\w*[^\-]\.cow" | shuf -n 1)
 elif [ -n "$NATIONAL_DEX_FORMS" ]; then
-	POKEMON_COW=$(find $POKEMON_PATH -name "${NATIONAL_DEX_FORMS}_*.cow" | shuf -n 1)
+	POKEMON_COW=$(find $POKEMON_PATH -printf "%f" -name "${NATIONAL_DEX_FORMS}_*.cow" | shuf -n 1)
 elif [ -n "$COW_FILE" ]; then
 	POKEMON_COW="$COW_FILE"
 else
-	POKEMON_COW=$(find $POKEMON_PATH -name "*.cow" | shuf -n 1)
+	POKEMON_COW=$(find $POKEMON_PATH -printf "%f" -name "*.cow" | shuf -n 1)
 fi
 
 # Get the pokemon name.
